tests/integration/security/reachability/context.go

```diff
@@ -0,0 +1,171 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reachability
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+type TestCase struct {
+	ConfigFile string
+	Namespace  namespace.Instance
+
+	RequiredEnvironment environment.Name
+
+	// Indicates whether a test should be created for the given configuration.
+	Include func(src echo.Instance, opts echo.CallOptions) bool
+
+	// Handler called when the given test is being run.
+	OnRun func(ctx framework.TestContext, src echo.Instance, opts echo.CallOptions)
+
+	// Indicates whether the test should expect a successful response.
+	ExpectSuccess func(src echo.Instance, opts echo.CallOptions) bool
+}
+
+// Context is a context for reachability tests.
+type Context struct {
+	ctx                   framework.TestContext
+	g                     galley.Instance
+	p                     pilot.Instance
+	Namespace             namespace.Instance
+	A, B, Headless, Naked echo.Instance
+}
+
+// CreateReachabilityContext creates and initializes reachability context.
+func CreateContext(ctx framework.TestContext, g galley.Instance, p pilot.Instance) Context {
+	ns := namespace.NewOrFail(ctx, ctx, namespace.Config{
+		Prefix: "reachability",
+		Inject: true,
+	})
+
+	var a, b, headless, naked echo.Instance
+	echoboot.NewBuilderOrFail(ctx, ctx).
+		With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+		With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+		With(&headless, util.EchoConfig("headless", ns, false, nil, g, p)).
+		With(&naked, util.EchoConfig("naked", ns, false, echo.NewAnnotations().
+			SetBool(echo.SidecarInject, false), g, p)).
+		BuildOrFail(ctx)
+
+	return Context{
+		ctx:       ctx,
+		g:         g,
+		p:         p,
+		Namespace: ns,
+		A:         a,
+		B:         b,
+		Headless:  headless,
+		Naked:     naked,
+	}
+}
+
+// Run runs the given reachability test cases with the context.
+func (rc *Context) Run(testCases []TestCase) {
+	callOptions := []echo.CallOptions{
+		{
+			PortName: "http",
+			Scheme:   scheme.HTTP,
+		},
+		{
+			PortName: "http",
+			Scheme:   scheme.WebSocket,
+		},
+		{
+			PortName: "tcp",
+			Scheme:   scheme.HTTP,
+		},
+		{
+			PortName: "grpc",
+			Scheme:   scheme.GRPC,
+		},
+	}
+
+	for _, c := range testCases {
+		c := c
```

Comment:
what's this for?

Reply:
don't know :). It was there before. Anyway, I agree it has no value, so removed.